src/init/documentHandlers.ts

```diff
@@ -51,3 +54,22 @@ function selectionChangeCallback(
     }
   }
 }
+
+function contextMenuAnnotateCallback(request, sender) {
+  if (request.action === 'ANNOTATE') {
+    /*
+     * For now, do not check for being inside article.
+     * Reason: checking ContextMenu API selection for being insideArticle is possible, but uncomfortable,
+     * as context menu actions are handled in the separate background script.
+     */
+    const selection = handlers.selector.captureDocumentSelection();
+
+    if (selection.length === 1) {
+      store.dispatch(setSelectionRange(selection[0]));
+      const selectionCenter = handlers.selector.currentSingleSelectionCenter();
+      store.dispatch(showEditorAnnotation(selectionCenter.x, selectionCenter.y));
```

Comment:
Does it effectively mean that the editor appears in more or less the same place as it happens if it is invoked _normally_ (using "add annotation") mode?

Reply:
As it happens (rather to my surprise), the window selection's bounding box can be straightforwardly accessed.
From context menu, Editor opens at the center of the selection. I found it good enough. For now, in annotation mode it opens at the mouse location.
Yes, it's slightly inconsistent.
Which way is better? I think both are good enough and the inconsistency itself doesn't compromise the usability. So I think making it consistent is a matter for another sprint.

Reply:
So more or less the same behavior - I am perfectly happy with that :smile: